src/Database/Query.php

```diff
@@ -37,7 +37,7 @@ class Query implements ExpressionInterface, IteratorAggregate
     /**
      * Connection instance to be used to execute this query.
      *
-     * @var \Cake\Datasource\ConnectionInterface
+     * @var \Cake\Datasource\ConnectionInterface|\Cake\Database\Connection
```

Comment:
Given the fact that 3.5 is basically there now, I reverted mine :) Thank you for doing the fixes!